modules/framework/galasa-parent/dev.galasa.framework.api.runs/src/main/java/dev/galasa/framework/api/runs/routes/GroupRunsRoute.java

```diff
@@ -83,4 +89,68 @@ public HttpServletResponse handlePostRequest(
         return getResponseBuilder().buildResponse(request, response, "application/json", gson.toJson(scheduleStatus), HttpServletResponse.SC_CREATED);
     }
 
+    @Override
+    public HttpServletResponse handlePutRequest(String groupName, HttpRequestContext requestContext,
+            HttpServletResponse res) throws ServletException, IOException, FrameworkException {
+
+        HttpServletRequest request = requestContext.getRequest();
+        String strippedGroupName = groupName.substring(1); //remove leading slash
+
+        GroupRunActionJson runAction = getRunActionFromRequestBody(request);
+        String responseBody = updateRunStatus(strippedGroupName, runAction);
+
+        int responseStatusCode = HttpServletResponse.SC_ACCEPTED;
+        
+        if (responseBody.isBlank()) {
+            responseStatusCode = HttpServletResponse.SC_OK;
+            responseBody = String.format(GAL5430_GROUP_RUNS_ALREADY_FINISHED.toString(), strippedGroupName);
+        }
+
+        return getResponseBuilder().buildResponse(request, res, "text/plain", responseBody, responseStatusCode);
+    }
+
+    private String updateRunStatus(String groupId, GroupRunActionJson runAction)
+            throws InternalServletException, ResultArchiveStoreException {
+
+        String responseBody = "";
+        List<IRun> groupedRuns = getRuns(groupId);
+        GroupRunActionStatus cancelAction = GroupRunActionStatus.getfromString(runAction.getResult());
```

Comment:
I think I'm finding it hard to follow why we are essentially storing the result of "cancelled" in two different things - the `GroupRunActionJson` then it seems like we just transfer it to the `GroupRunActionStatus` - can you explain please why we need both?